src/rust/engine/fs/store/src/lib.rs

```diff
@@ -698,9 +698,19 @@ impl Store {
     let store = self.clone();
     async move {
       if let RootOrParentMetadataBuilder::Root(..) = root_or_parent_metadata {
-        fs::safe_create_dir_all(&destination)?;
+        let destination = destination.clone();
+        store
+          .local
+          .executor()
+          .spawn_blocking(move || fs::safe_create_dir_all(&destination))
```

Comment:
I suspect this could go even further by pre-processing a list of all directories that will be made, and making each one exactly once (with `create_dir`), rather than `create_dir_all`ing them all as we go, and repeatedly making the same intermediate dirs.

Reply:
We only use `create_dir_all` for the `Root`: other directories are created with `create_dir`.